### PR TITLE
Simplifying reachability.

### PR DESCRIPTION
**Pull request by lvella**
Had two collections holding the same thing. Now have one.

### CODE REVIEW

File: riscv/src/reachability.rs [hunk 1]
```diff
@@ -10,14 +10,13 @@ pub fn filter_reachable_from(
 ) {
     let replacements = extract_replacements(statements);
     let label_offsets = extract_label_offsets(statements);
-    let mut queued_labels: BTreeSet<&str> = vec![label].into_iter().collect();
-    let mut referenced_labels: BTreeSet<&str> = vec![label].into_iter().collect();
+    let mut queued_labels = BTreeSet::from([label]);
+    let mut referenced_labels = BTreeSet::from([label]);
```
Comment (leonardoalt) [inline riscv/src/reachability.rs:14]: This seems unused?

Comment (lvella) [reply riscv/src/reachability.rs:14]: It is used. Please expand the code until the end of the function.

File: riscv/src/reachability.rs [hunk 2]
```diff
@@ -49,8 +48,7 @@ pub fn filter_reachable_from(
             panic!();
         };
         for referenced in &new_references {
-            if !queued_labels.contains(referenced) && !processed_labels.contains(referenced) {
-                label_queue.push(referenced);
+            if !processed_labels.contains(referenced) {
```
Comment (leonardoalt) [inline riscv/src/reachability.rs:51]: If you don't check that is already in `queued_labels` you might insert it twice.

Comment (leonardoalt) [reply riscv/src/reachability.rs:51]: Before the first instance of the same item is visited I mean.

Comment (lvella) [reply riscv/src/reachability.rs:51]: It is a Set, there can be only one.

**Review by leonardoalt — APPROVED** [review]

File: riscv/src/reachability.rs [hunk 3]
```diff
@@ -10,14 +10,13 @@ pub fn filter_reachable_from(
...
-    let mut referenced_labels: BTreeSet<&str> = vec![label].into_iter().collect();
+    let mut queued_labels = BTreeSet::from([label]);
+    let mut referenced_labels = BTreeSet::from([label]);
     let mut processed_labels = BTreeSet::<&str>::new();
     // Labels that are included in a basic block that starts with a different label,
     // or object labels.
     let mut secondary_labels = BTreeSet::<&str>::new();
-    let mut label_queue = vec![label];
-    while let Some(l) = label_queue.pop() {
+    while let Some(l) = queued_labels.pop_first() {
```
Comment (chriseth) [inline riscv/src/reachability.rs:19]: `pop_first` on `BTreeSet` is marked as unstable for me - why did the tests pass?

Comment (chriseth) [reply riscv/src/reachability.rs:19]: Ah, it seems a new stable got released since I installed :)

**Pull request merged**